src/platform/intel/cavs/lib/clk.c

```diff
@@ -42,7 +43,7 @@ static inline void select_cpu_clock_hw(int freq_idx, bool release_unused)
 
 	/* wait for requested clock to be on */
 	while ((io_reg_read(SHIM_BASE + SHIM_CLKSTS) &
-		status_mask) != status_mask)
+		status_mask) != status_mask && --count)
```

Comment:
Where do we trace_err() about the timeout if it happens ?

Reply:
...and not only trace_err but also... _not change to that clock_.

Reply:
updated now.